src/Microsoft.Dnx.Runtime/PackageFilesSource.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Dnx.Runtime.Json;
+
+namespace Microsoft.Dnx.Runtime
+{
+    public class PackageFilesSource
+    {
+        public IReadOnlyList<string> Values { get; }
+        public int Line { get; }
+        public int Column { get; }
+
+        internal PackageFilesSource(JsonValue json)
+            : this(ExtractValues(json), json.Line, json.Column)
+        {
+        }
+
+        public PackageFilesSource(IEnumerable<string> values, int line, int column)
+        {
+            Values = values.ToList().AsReadOnly();
```

Comment:
Do all callers call an IList<string> directly? If so change the input to list or array or something specific. This isn't a generic type that we want to expose to people is it?